### DISCUSSION TITLE
Rendering blemishes about triton ocean on osgearth

### DISCUSSION THREAD

**Discussion by pskpg** [pelicanmapping/osgearth#2438] [Support Q&A] [answered]
Hi, Gwaldron and Frank!
I've been working on osgearth & triton for a long time. Now I've got an overall usable ocean on the earth. But some rendering  blemishes still exist, as shown in the picture:
![0](https://github.com/gwaldron/osgearth/assets/20101178/8b0f5dae-64cc-4dff-a6fa-25d13bb912a9)
There are two rendering error in the picture: 1. a "lake" presents upon the moutain, which should not be there; 2. a white strip presents at the foot position throughout the whole mountain. 
![0_labeled](https://github.com/gwaldron/osgearth/assets/20101178/eeba78b8-4b94-488e-a202-9a44f8541862)
Through some tests, I found the first error is caused by the small near far ratio — when I adjust the near far ratio to about 10exp(-1.3~-1.4), the lake fades away, like this:
![1](https://github.com/gwaldron/osgearth/assets/20101178/cfbc8b62-7f43-4a3d-8908-c4c13b8d10f6)
About the second error (the white strip at the moutain foot), I guess something must be wrong with the depth order or rendering order of the ocean and earth. Here are two vedios demonstrate the errors:
[vedio 1](https://github.com/gwaldron/osgearth/assets/20101178/350a4829-f168-46f3-b034-0cfa8f9ffcf3)
[vedio 2](https://github.com/gwaldron/osgearth/assets/20101178/16a5fd3b-aa55-4011-8c67-6655432fc456)
Could you please give me some suggestions to fix it?

Thanks!

**Comment by gwaldron** [comment, marked as answer]
Are you using the osgEarth logarithmic depth buffer? (You might be using it by default if you use the MapHelper class.)
If so, make sure you are installing the Triton user shaders found in `src/osgEarthTriton/shaders`.  There's a README file in there.
